### PR TITLE
Improve otaindexer

### PR DESCRIPTION
**Pull request by 0xBahamoot**
Improve otaindexer

### CODE REVIEW

File: otaindexer/master.go [hunk 1]
```diff
@@ -477,7 +477,17 @@ func ReCheckOTAKey(otaKey, pubKey string, reIndex bool) error {
...
+		if len(data.NFTIndex) == 0 {
+			data.NFTIndex = make(map[string]shared.CoinInfo)
+		}
+		if len(data.CoinIndex) == 0 {
+			data.CoinIndex = make(map[string]shared.CoinInfo)
+		}
 		if coin.IsNFT {
+			if len(data.NFTIndex) == 0 {
+				data.NFTIndex = make(map[string]shared.CoinInfo)
+			}
```
Comment (0xakk0r0kamui) [inline otaindexer/master.go:490]: redundant?

File: otaindexer/utils.go [hunk 2]
```diff
@@ -45,6 +45,12 @@ func doesCoinBelongToKeySet(c *coin.CoinV2, keySet *incognitokey.KeySet, tokenID
 	}
 	if assetTag == nil {
 		tokenID = common.PRVCoinID.String()
+	} else {
+		// if assetTag.String() == shared.AccessTokenAssetTag {
+		// 	tokenID = common.PdexAccessCoinID.String()
+		// 	isNFT = true
+		// 	return pass, tokenID, rK, isNFT
+		// }
```
Comment (0xakk0r0kamui) [inline otaindexer/utils.go:53]: redundant?

File: otaindexer/master.go [hunk 3]
```diff
@@ -505,6 +515,9 @@ func ReCheckOTAKey(otaKey, pubKey string, reIndex bool) error {
 				data.NFTIndex[coin.RealTokenID] = cidx
 			}
 		} else {
+			if len(data.CoinIndex) == 0 {
+				data.CoinIndex = make(map[string]shared.CoinInfo)
+			}
```
Comment (0xakk0r0kamui) [inline otaindexer/master.go:520]: redundant?

File: otaindexer/utils.go (hunk 2 above)
Comment (0xBahamoot) [reply otaindexer/utils.go:53]: this is pre-code for nextota

File: otaindexer/master.go (hunk 1 above)
Comment (0xBahamoot) [reply otaindexer/master.go:490]: just to make sure that map is initialized

File: otaindexer/master.go (hunk 3 above)
Comment (0xBahamoot) [reply otaindexer/master.go:520]: just to make sure that map is initialized

File: otaindexer/worker_new.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+	var wg sync.WaitGroup
+	tempOTACoinsCh := make(chan map[string]shared.CoinData, shared.ServiceCfg.MaxConcurrentOTACheck)
+	for idx, c := range coinList {
+		wg.Add(1)
+		go func(cn shared.CoinData) {
+			newCoin := new(coin.CoinV2)
+			err := newCoin.SetBytes(cn.Coin)
+			if err != nil {
+				panic(err)
```
Comment (0xakk0r0kamui) [inline otaindexer/worker_new.go:28]: log error or defer wg.Done()

File: otaindexer/worker_new.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+					if len(totalTxs[key][coin.RealTokenID]) == 0 {
+						totalTxs[key][coin.RealTokenID] = make(map[string]struct{})
+					}
+					totalTxs[key][coin.RealTokenID][coin.TxHash] = struct{}{}
+				}
+			}
+			tempOTACoinsCh = make(chan map[string]shared.CoinData, shared.ServiceCfg.MaxConcurrentOTACheck)
+		}
+	}
+	close(tempOTACoinsCh)
```
Comment (0xakk0r0kamui) [inline otaindexer/worker_new.go:136]: `if (idx+1)%shared.ServiceCfg.MaxConcurrentOTACheck == 0 || idx+1 == len(coinList) {` => close tempOTACoinsCh twice => panic
`

**Pull request closed without merging**